docbrown/db/benches/base.rs

```diff
@@ -56,93 +58,80 @@ fn load_csv(graph: &mut GraphDB, path: &Path, source: usize, target: usize, time
     }
 }
 
-pub fn additions(c: &mut Criterion) {
-    let mut graph = GraphDB::new(4);
-    graph.add_vertex(0, 0, &vec![]);
-
-    let mut g = c.benchmark_group("additions");
-    g.throughput(Throughput::Elements(1));
-    g.warm_up_time(Duration::from_millis(10));
-    g.measurement_time(Duration::from_millis(10));
+pub fn ingestion(c: &mut Criterion) {
+    fn make_index_gen() -> Box<dyn Iterator<Item = u64>> {
+        let mut rng = rand::thread_rng();
+        let range = Uniform::new(u64::MIN, u64::MAX);
+        Box::new(rng.sample_iter(range))
+    }
+    fn make_time_gen() -> Box<dyn Iterator<Item = i64>> {
+        let mut rng = rand::thread_rng();
+        let range = Uniform::new(i64::MIN, i64::MAX);
+        Box::new(rng.sample_iter(range))
+    }
 
-    let mut times: Range<i64> = 0..i64::MAX;
-    let mut indexes: Range<u64> = 0..u64::MAX;
+    let mut indexes = make_index_gen();
+    let mut times = make_time_gen();
+    let mut index_sample = || indexes.next().unwrap();
```

Comment:
can't these things be just `fn index_sample` ?

Reply:
If I change it to be `fn`, it doesn't compile because it cannot capture `indexes` from the context. Or do you mean doing something else?

Reply:
Interesting, ok